### PR TITLE
don't use fixed header

### LINKED ISSUE #7484: Links to header cells hide the cell under the menu
Here's what things look like before and after clicking on link in the header cell.

before-after

### PR DESCRIPTION
**Pull request by minrk**
instead, use static header and flexbox to accomplish the same thing.

fixes various issues related to scrolling, anchors, etc.

Just a proposal for discussion, not necessarily the right thing. I haven't found a problem yet, though.

closes #7484

### CODE REVIEW

**Comment by jdfreder** [comment]
@ellisonbg here's @minrk 's solution, but it doesn't look like it reintroduces the layout manager.

File: IPython/html/templates/page.html [hunk 1]
```diff
@@ -106,8 +106,6 @@
   {% endblock %}
 </div>
 
-<div id="header-spacer"></div>
```
Comment (jdfreder) [inline IPython/html/templates/page.html]: You'll also want to delete the accompanying Javascript which sets the height of the spacer and all references to it (see `Page.prototype._resize_header` in `/static/base/js/page.js`).

**Comment by jdfreder** [comment]
Seems okay so far (testing on Chrome)...

**Comment by ellisonbg** [comment]
We definitely need to test:

* Printing multipage notebooks from the browser.
* Performance problems that are fixed in master.
* Pager operation

On Wed, Jan 21, 2015 at 12:45 PM, Jonathan Frederic <
notifications@github.com> wrote:

> Seems okay so far (testing on Chrome)...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ipython/ipython/pull/7517#issuecomment-70917931>.
>



-- 
Brian E. Granger
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by jdfreder** [comment]
Printing doesn't work

**Comment by minrk** [comment]
Printing is fixed.

**Comment by jdfreder** [comment]
Awesome!  I tested and it seems to work.  Okay, this satisfies me now :)

**Comment by minrk** [comment]
repaint events appear identical before/after this PR.

**Comment by minrk** [comment]
Thanks, @jdfreder, you've fought with this stuff the most recently, so I trust you to remember what the likely failures are.

**Comment by minrk** [comment]
@jdfreder thanks, shall we merge, then?

**Comment by jdfreder** [comment]
> @jdfreder thanks, shall we merge, then?

I think so

**Comment by minrk** [comment]
@jdfreder thanks, now we'll see what horrible thing I've done.

**Comment by Carreau** [comment]
Yeah, the header definitively scroll out of view now...

**Pull request merged**